lib/js/components/Toast/Toast.vue

```diff
@@ -236,30 +195,31 @@ export default {
 				? BUTTON_COLORS.DANGER
 				: BUTTON_COLORS.NEUTRAL;
 		},
-		disableTeleport() {
-			return this.appendTo === null || this.appendToElement === this.$parent.$el;
-		},
 	},
 	watch: {
-		appendTo() {
-			this.choseAppendToElement();
-			this.calculateStyles();
-			this.resizeObserver?.disconnect();
-			this.resizeObserver?.observe(this.appendToElement);
-			//We watch the toast's card height because it changes by 1px at the beginning, and we don't know why
-			this.resizeObserver?.observe(this.$refs?.toastCard);
-		},
-		position() {
-			this.calculateStyles();
+		styles: {
+			handler() {
+				const { right, left } = this.$el.getClientRects()[0];
+				if (parseInt(left, 10) < 0 || right > document.documentElement.clientWidth) {
+					this.styles = {
+						left: '50%',
+						transform: 'translateX(-50%)',
+						bottom: `${TOAST_OFFSET}px`,
+					};
+				}
+			},
 		},
+		flush: 'post',
 	},
 	mounted() {
-		this.choseAppendToElement();
+		this.setBoundariesSelectorElement();
 		this.calculateStyles();
-
-		this.resizeObserver.observe(this.appendToElement);
-		//We watch the toast's card height because it changes by 1px at the beginning, and we don't know why
-		this.resizeObserver?.observe(this.$refs?.toastCard);
+		if (this.boundariesSelectorElement) {
+			this.boundariesSelectorElementResizeObserver = new ResizeObserver(() => {
```

Comment:
should we use observer instead of [onresize](https://developer.mozilla.org/en-US/docs/Web/API/Window/resize_event)?

Reply:
The case is that window size can be changed without changing container size, and the opposite is true as well